### PR TITLE
Monitoring Data Collector Launch and Exit events.

### PR DESCRIPTION
**Pull request by harshjain2**
Monitoring Data Collector Launch and Exit events, logging error in case data collector crashes.

### CODE REVIEW

File: src/Microsoft.TestPlatform.Common/ProcessCallbacks.cs [hunk 1]
```diff
@@ -1,17 +1,18 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-namespace Microsoft.TestPlatform.TestHostProvider.Hosting
+namespace Microsoft.VisualStudio.TestPlatform.Common
```
Comment (codito) [inline src/Microsoft.TestPlatform.Common/ProcessCallbacks.cs]: I am not sure if Common should include utility code. Isn't coreutilities/platformabstractions the component?

File: src/Microsoft.TestPlatform.Common/ProcessCallbacks.cs [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
+
     using Microsoft.VisualStudio.TestPlatform.ObjectModel;
     using Microsoft.VisualStudio.TestPlatform.ObjectModel.Host;
     using Microsoft.VisualStudio.TestPlatform.ObjectModel.Logging;
     using Microsoft.VisualStudio.TestPlatform.PlatformAbstractions.Interfaces;
 
-    internal class TestHostManagerCallbacks
+    internal class ProcessCallbacks
     {
         public static void ErrorReceivedCallback(StringBuilder testHostProcessStdError, string data)
```
Comment (codito) [inline src/Microsoft.TestPlatform.Common/ProcessCallbacks.cs]: If it's a generic API, the names should change.

File: src/Microsoft.TestPlatform.Common/ProcessCallbacks.cs (hunk 1 above)
Comment (harshjain2) [inline src/Microsoft.TestPlatform.Common/ProcessCallbacks.cs]: This class has dependency on `HostProviderEventArgs` which is in object model, therefore, I intentionally kept it in Common. 
Ideally it should go to platform abstractions, could be along with ProcessHelper

Comment (codito) [inline src/Microsoft.TestPlatform.Common/ProcessCallbacks.cs]: HostProviderEventArgs is interesting for DataCollectors? Can that not stay along side TestHostProvider implementation?

**Review by codito — CHANGES_REQUESTED** [review]
We need to figure a better place for the callbacks.

File: src/Microsoft.TestPlatform.Common/ProcessCallbacks.cs (hunk 1 above)
Comment (harshjain2) [inline src/Microsoft.TestPlatform.Common/ProcessCallbacks.cs]: @codito I don't think, except for the name there is anything in `HostProviderEventArgs` that is specific to TestHostProvider.

**Pull request closed without merging**